src/components/sharing/ShareButton/index.js

```diff
@@ -0,0 +1,134 @@
+"use strict";
+
+var _                       = require('lodash');
+var React                   = require('react/addons');
+var I18nMixin               = require('../../mixins/I18n');
+var Icon                    = require('../../helpers/Icon');
+var ShareBox                = require('../ShareBox');
+var format                  = require('../../../lib/format');
+var ReactCSSTransitionGroup = React.addons.CSSTransitionGroup;
+
+var serviceConfigs = [
+  {
+    name: "facebook",
+    url: "http://www.facebook.com/sharer.php?u={url}"
+  },
+  {
+    name: "twitter",
+    url: "https://twitter.com/share?url={url}&text={title}"
+  },
+  {
+    name: "googleplus",
+    url: "https://plus.google.com/share?url={url}",
+    icon: "google-plus"
+  },
+  {
+    name: "pinterest",
+    url: "https://pinterest.com/pin/create/bookmarklet/?media={img}&url={url}&description={title}"
+  }
+];
+
+module.exports = React.createClass({
+  mixins: [I18nMixin],
+  displayName: "ShareButton",
+  propTypes: {
+    buttons: React.PropTypes.array,
+    shareUrl: React.PropTypes.string,
+    shareTitle: React.PropTypes.string,
+    shareImage: React.PropTypes.string,
+    renderIcon: React.PropTypes.bool,
+    i18n: React.PropTypes.object
+  },
+
+  getDefaultProps: function() {
+    return {
+      buttons: [
+        'facebook',
+        'twitter',
+        'googleplus',
+        'pinterest'
+      ],
+      shareUrl: window.location.href,
+      shareTitle: document.title,
+      shareImage: '',
+      renderIcon: true,
+      defaultI18n: {
+        buttonLabel: 'Share this page',
+        shareLinkLabel: 'Share a direct link',
+        shareViaLabel: 'Share via'
+      }
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      active: false
+    };
+  },
+
+  toggleActive: function() {
+    this.setState({ active: !this.state.active });
+  },
+
+  filterServices: function() {
+    var filteredServices = [];
+
+    _.forEach(this.props.buttons, function(name) {
+      filteredServices.push(_.findWhere(serviceConfigs, { 'name': name }));
+    });
+
+    return filteredServices;
+  },
+
+  formatServiceUrls: function(services) {
+    return _.forEach(services, function(service) {
+      service.url = format(service.url, {
+        "url": this.props.shareUrl,
+        "title": this.props.shareTitle,
+        "img": this.props.shareImage
+      }, true);
+    }, this);
+  },
+
+  renderIcon: function() {
+    if (this.props.renderIcon) {
+      return <Icon className="ShareButton__icon" icon="share" />;
+    }
+  },
+
+  renderShareBox: function() {
+    if (this.state.active) {
+      var props    = this.props;
+      var image    = props.shareImage;
+      var services = this.filterServices();
+
+      this.formatServiceUrls(services);
```

Comment:
Unfortunately, there's a potential problem if for some reason you use more than one ShareBox on a page, they share the same list of `serviceConfigs` above (although filtered) and you are updating the global list in place. Perhaps make formatServiceUrls() not update in place but use _.cloneDeep(services) and update the copy, or just use `_.map(services, function(service) { return { name: service.name, icon: service.icon, url: format(service.url, urlParams, true) }; }` and define urlParams outside the loop as it doesn't change. Then this line would be `services = this.formatServices(services);` ?